### ISSUE TITLE
Opening desktop client intercepted by proxy

### ISSUE THREAD

**Issue by gpanders** [zadam/trilium#577]
Trilium version: 0.32.4
OS: macOS

I know the title is a bit vague but here is the issue I'm having:

At work I am behind a corporate proxy that intercepts all HTTP requests. I have a Trilium instance set up on a remote server that I want to connect to using the desktop client. I can access the server instance in my browser and I can also access it when I am not on my work network.

The problem is that when I open the desktop application, I get the following error (which comes from the corporate proxy):

```
The following error was encountered while trying to retrieve the URL: http://localhost.mycompany.com:61897/

           Connection to ::1 failed.

The system returned: (111) Connection refused
```

Note that the proxy **does not** block my instance, since I can still access it from my web browser. HTTP requests to `localhost` do not go through the proxy, but for some reason `localhost` is being translated into `localhost.mycompany.com` which _is_ being intercepted by the proxy.

This is on macOS and I do not have the same issue on Windows.

1. Is there a way I can force the desktop client to connect directly to my server instance instead of trying to connect to localhost?
2. Is it possible to "fix" the localhost connection so that the address resolution actually resolves to `localhost` and not `localhost.mycompany.com`?

Here are the contents of the log file:
```
10:32:28.745 INFO  Nothing to push
10:32:31.465 INFO  Finished pull
10:32:31.466 INFO  Nothing to push
10:32:35.388 INFO  Content hash computation took 5ms
10:32:35.388 INFO  Content hash checks PASSED
```
(repeated many times)

**Comment by zadam** [comment]
Hello, this is weird. Could you try to use latest 0.33.3 - I changed from using localhost to access the backend to 127.0.0.1 - the difference is that the latter won't go through name resolution. Not sure if this helps, but worth a try.

**Comment by gpanders** [comment]
That does seem to fix it actually -- working again in 0.33.3.

Thanks!
